### PR TITLE
ITKDev - favorites list

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5396

#### Description

Favorites list module 
**Will remove react/design branches before merge**

#### Screenshot of the result

image

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ];
+
+    $form['settings']['page_size_mobile'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Page size mobile'),
+      '#default_value' => $config->get('page_size_mobile') ?? 25,
+      '#min' => 1,
+      '#step' => 1,
+    ];
+    $form['settings']['page_size_desktop'] = [
```
Comment (cableman) [inline web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:48]: Maybe an empty line between these two.

**Review by cableman — CHANGES_REQUESTED** [review]

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer
+  ) {
+  }
```
Comment (spaceo) [inline web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php:28]: Is it on purpose that the constructor is empty? If yes please explain in a comment why.

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    $form['settings'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Basic settings'),
+      '#tree' => FALSE,
+    ];
+
+    $form['settings']['page_size_mobile'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Page size mobile'),
+      '#default_value' => $config->get('page_size_mobile') ?? 25,
```
Comment (spaceo) [inline web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:44]: Not fond of hardcoding values on the fly. At least it should be possible to export the config and make it a part of this PR.

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      '#min' => 1,
+      '#step' => 1,
+    ];
+
+    $form['settings']['page_size_desktop'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Page size desktop'),
+      '#default_value' => $config->get('page_size_desktop') ?? 25,
```
Comment (spaceo) [inline web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:52]: Same as previous comment

File: web/modules/custom/dpl_favorites_list/src/Plugin/Block/FavoritesListBlock.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      // Page size.
+      "page-size-desktop" => $favoritesListSettings->get('page_size_desktop'),
+      "page-size-mobile" => $favoritesListSettings->get('page_size_mobile'),
+
+      // Texts.
+      'group-modal-checkbox-text' => $this->t("Choose all renewable", [], ['context' => 'Favorites list']),
+      "favorites-list-materials-text" => $this->t("@count materials", [], ['context' => 'Favorites list']),
+      "favorites-list-header-text" => $this->t("Favorites", [], ['context' => 'Favorites list']),
+      "by-author-text" => $this->t("By", [], ['context' => 'Favorites list']),
+      "et-al-text" => $this->t("...", [], ['context' => 'Favorites list']),
```
Comment (spaceo) [inline web/modules/custom/dpl_favorites_list/src/Plugin/Block/FavoritesListBlock.php:93]: The translation key to this translation would then be: "..." which you need to search for as an admin in order to translate it. Is it intentional?

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks god. Have a few comments

File: web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php:28]: Constructor Property Promotion is a new syntax in PHP 8. So there is nothing to comment on here.

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php (hunk 3 above)
Comment (cableman) [reply web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:44]: This is currently following the same pattern as in the other PR's that have been approved and merged.

File: web/modules/custom/dpl_favorites_list/src/Plugin/Block/FavoritesListBlock.php (hunk 5 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_favorites_list/src/Plugin/Block/FavoritesListBlock.php]: This would be a copy paste error, and is fixed :)

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php (hunk 3 above)
Comment (spaceo) [reply web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:44]: I understand that you are under time pressure and choose to ignore my request.

File: web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php (hunk 4 above)
Comment (spaceo) [reply web/modules/custom/dpl_favorites_list/src/Form/FavoritesListSettingsForm.php:52]: I understand that you are under time pressure and choose to ignore my request.

File: web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php (hunk 2 above)
Comment (spaceo) [reply web/modules/custom/dpl_favorites_list/src/Controller/DplFavoritesListController.php:28]: Ah I se 👍

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**